pkg/agent/util/net_linux.go

```diff
@@ -376,3 +378,148 @@ func renameHostInterface(oriName string, newName string) error {
+		return "", false, fmt.Errorf("failed to create internal port: %v", err)
+	}
+
+	// Wait a few seconds for OVS bridge local port.
+	if err := wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+		link, err := netlink.LinkByName(interfaceName)
+		if err != nil {
+			klog.V(4).InfoS("OVS bridge local port is not ready", "port", interfaceName, "err", err)
+			return false, nil
+		}
+		klog.InfoS("OVS bridge local port is ready", "type", link.Type(), "attrs", link.Attrs())
+		return true, nil
+	}); err != nil {
+		return "", false, fmt.Errorf("failed waiting for internal port to show up: %v", err)
+	}
+
+	localLink, err := netlink.LinkByName(interfaceName)
+	if err != nil {
+		return "", false, err
+	}
+	if _, _, err = SetLinkUp(interfaceName); err != nil {
+		return "", false, fmt.Errorf("failed to set link up: %v", err)
+	}
+
+	// Check if interface is configured with an IPv6 address: if it is, we need to ensure that IPv6
+	// is enabled on the OVS internal port as we need to move all IP addresses over.
+	for _, ip := range interfaceIPs {
+		if ip.IP.To4() == nil {
+			klog.InfoS("Interface has IPv6 address, ensuring that IPv6 is enabled on bridge local port", "port", interfaceName)
+			if err := EnsureIPv6EnabledOnInterface(interfaceName); err != nil {
+				klog.ErrorS(err, "Failed to ensure that IPv6 is enabled on bridge local port, moving uplink IPs to bridge is likely to fail", "port", interfaceName)
+			}
+			break
+		}
+	}
+
+	if err = ConfigureLinkAddresses(localLink.Attrs().Index, interfaceIPs); err != nil {
+		return "", false, err
+	}
+	if err = ConfigureLinkAddresses(interfaceIndex, nil); err != nil {
+		return "", false, err
+	}
+	// Restore the host routes which are lost when moving the network configuration of the
+	// host interface to OVS bridge interface.
+	if err = ConfigureLinkRoutes(localLink, interfaceRoutes); err != nil {
+		return "", false, err
+	}
+	return bridgedName, false, nil
+}
+
+// RestoreHostInterfaceConfiguration restore the configuration from bridge back to host interface, reverting the
+// actions taken in PrepareHostInterfaceConnection.
+func RestoreHostInterfaceConfiguration(brName string, interfaceName string) {
+	klog.InfoS("Restoring bridge config to host interface")
+	if interfaceName != "" {
+		bridgedName := GenerateUplinkInterfaceName(interfaceName)
+		// restore if interface eth0~ exists
+		if exists, err := interfaceExists(bridgedName); exists || err != nil {
+			// get interface config
+			var interfaceIPs []*net.IPNet
+			var interfaceRoutes []interface{}
+			if exists, err = interfaceExists(interfaceName); exists || err != nil {
+				_, interfaceIPs, interfaceRoutes, err = GetInterfaceConfig(interfaceName)
+				if err != nil {
+					klog.ErrorS(err, "Failed to get interface config", "interface", interfaceName)
+				}
+			}
+
+			// remove host interface (eth0~) from bridge
+			if err = deleteOVSPort(brName, bridgedName); err != nil {
```

Comment:
Good to check exists and skip this call.

Reply:
This is nested under the block if both host interface and bridged interfaces exist, lmk if anything else is required.

Reply:
The current version LGTM.